util/include/Handoff.hpp

```diff
@@ -0,0 +1,94 @@
+// Concord
+//
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").  You may not use this product except in
+// compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license terms. Your use of
+// these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include <queue>
+#include <thread>
+#include <mutex>
+#include <atomic>
+#include <condition_variable>
+#include <functional>
+#include <exception>
+#include "Logger.hpp"
+
+namespace concord::util {
+/**
+ * Simple handing off to a dedicated thread functionality.
+ * Implemented by passing a function with a trivial signature void(void).
+ * Parameters are bound to an arbitrary function by a caller.
+ */
+class Handoff {
+  typedef std::lock_guard<std::mutex> guard;
+  typedef std::function<void()> func_type;
+
+ public:
+  Handoff() {
+    thread_ = std::thread([this] {
+      try {
+        for (;;) pop()();
```

Comment:
I think we should either specify `func_type` as `noexcept` or handle any exceptions, because `pop()()` calls user code that might throw.

Reply:
done